src/components/RadioInput/index.tsx

```diff
@@ -10,13 +10,10 @@ export const RadioInput: React.FC<{
   onChange?: (event: RadioInputEvent) => void;
 }> = (props) => {
   const { children, checked = false, onChange = () => null, ...rest } = props;
-  const [state, setState] = useState(checked);
 
   const handleClick = () => {
-    setState(!state);
-
     const event = {
-      target: { value: !state },
+      target: { value: checked },
```

Comment:
I feel like these changes may break other parts of the app, did you test this change?

It seems like the checked state will never be toggled, except you maintain state from outside the component